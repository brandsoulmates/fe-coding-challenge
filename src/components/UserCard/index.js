import React from "react"
import { Link } from "gatsby"

const UserCard = ({ user }) => {
  return (
    <div className="m-4 px-6 pt-6 pb-10 text-center rounded-xl shadow-lg border-2 border-blue-500 bg-white">
      {/* src should be replaced with {user.avatar_url} */}
      <img src={user.avatar_url} alt={user.login} className="mx-auto w-48 h-48 rounded-full" />
      <p className="mt-4 mb-6 text-2xl font-medium">{user.login}</p>
      <Link to={`/user/${user.login}`} className="px-5 py-3 text-white bg-sky-700 rounded-lg">
        VIEW PROFILE
      </Link>
    </div>
  )
}

export default UserCard
