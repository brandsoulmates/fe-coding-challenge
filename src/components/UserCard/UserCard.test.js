import React from "react"
import { render } from "@testing-library/react"

import UserCard from "."

describe('UserCard component', () => {
  it('renders the test correctly', () => {
    const mockUser = {
      login: 'gituser',
      avatar_url: 'https://avatars.githubusercontent.com/u/2?v=4'
    }
    const result = render(<UserCard user={mockUser} />)
    expect(result).toMatchSnapshot()
  })
})
