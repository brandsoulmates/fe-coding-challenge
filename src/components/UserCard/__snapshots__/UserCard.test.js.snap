// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard component renders the test correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="m-4 px-6 pt-6 pb-10 text-center rounded-xl shadow-lg border-2 border-blue-500 bg-white"
      >
        <img
          alt="gituser"
          class="mx-auto w-48 h-48 rounded-full"
          src="https://avatars.githubusercontent.com/u/2?v=4"
        />
        <p
          class="mt-4 mb-6 text-2xl font-medium"
        >
          gituser
        </p>
        <a
          class="px-5 py-3 text-white bg-sky-700 rounded-lg"
          href="/user/gituser"
        >
          VIEW PROFILE
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="m-4 px-6 pt-6 pb-10 text-center rounded-xl shadow-lg border-2 border-blue-500 bg-white"
    >
      <img
        alt="gituser"
        class="mx-auto w-48 h-48 rounded-full"
        src="https://avatars.githubusercontent.com/u/2?v=4"
      />
      <p
        class="mt-4 mb-6 text-2xl font-medium"
      >
        gituser
      </p>
      <a
        class="px-5 py-3 text-white bg-sky-700 rounded-lg"
        href="/user/gituser"
      >
        VIEW PROFILE
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
